strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/HostedRepositoryProvider.java

```diff
@@ -85,21 +78,19 @@ public OutputStream getOutputStreamInternal(RepositoryPath repositoryPath)
     @Override
     public List<Path> search(String storageId,
                              String repositoryId,
-                             Predicate predicate,
+                             RepositorySearchRequest predicate,
                              Paginator paginator)
     {
         List<Path> result = new LinkedList<Path>();
 
         Storage storage = configurationManager.getConfiguration().getStorage(storageId);
         Repository repository = storage.getRepository(repositoryId);
         
-        Selector<ArtifactEntry> selector = createSelector(storageId, repositoryId, predicate).with(paginator).fetch();
-        
-        QueryTemplate<List<ArtifactEntry>, ArtifactEntry> queryTemplate = new DetachQueryTemplate<>(entityManager);
-        
         RootRepositoryPath rootRepositoryPath = repositoryPathResolver.resolve(repository);
-        List<ArtifactEntry> searchResult = queryTemplate.select(selector);
-        for (ArtifactEntry artifactEntry : searchResult)
+        List<Artifact> searchResult = artifactIdGroupRepository.findArtifacts(storageId, repositoryId, predicate.getArtifactId(),
+                                                                              predicate.getCoordinateValues(),
+                                                                              paginator.getSkip(), paginator.getLimit());
```

Comment:
Cosmetic: When there are so many parameters, please have one per line as we have it across th rest of the codebase. It makes it much easier to read. Thanks! :)

```suggestion
        List<Artifact> searchResult = artifactIdGroupRepository.findArtifacts(storageId,
                                                                              repositoryId,
                                                                              predicate.getArtifactId(),
                                                                              predicate.getCoordinateValues(),
                                                                              paginator.getSkip(),
                                                                              paginator.getLimit());
```